File: github_code/tower-of-hanoi/cpp/54.c
#include <iostream>
#include "Algorithm/TowerOfHanoi.h"
using namespace std;

TowerOfHanoi::TowerOfHanoi(){
}
TowerOfHanoi::~TowerOfHanoi(){
}

void TowerOfHanoi::tower(int n, char BEG, char AUX, char END){
        //here disk 1<2<3....
        if(n==1){
            cout<<"Move disk "<<n<<" from "<<BEG<<" to "<<END<<"\n";
            return;
        }
        tower(n-1, BEG, END, AUX);
        cout<<"Move disk "<<n<<" from "<<BEG<<" to "<<END<<"\n";
        tower(n-1, AUX, BEG, END);
        return;
}

File: github_code_index/java/3/54.java
/*
TowerOfHanoi problem if we want to move N disks from source to destination, then
we first move N-1 disks from source to temp, then move the lowest Nth disk from source to
destination. Then will move N-1 disks from temp to destination
*/

package recursive;
public class TowerOfHanoi {

	public static void towerOfHanoi(int num, char src, char dst, char temp) {
		if (num < 1)
			return;
		
		towerOfHanoi(num - 1, src, temp, dst);
		System.out.println("Move " + num + " disk from peg " + src + " to peg " + dst);
		towerOfHanoi(num - 1, temp, dst, src);
	}

	public static void main(String[] args) {
		int num = 3;
		System.out.println("The sequence of moves in the Tower of Hanoi are : ");
		towerOfHanoi(num, 'A', 'C', 'B');
	}
}
